oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java

```diff
@@ -132,14 +132,23 @@ public void start() {
     @Override
     public void run() {
         while (true) {
-            ConsumerRecords<String, Bytes> consumerRecords = consumer.poll(Duration.ofMillis(500L));
-            if (!consumerRecords.isEmpty()) {
-                Iterator<ConsumerRecord<String, Bytes>> iterator = consumerRecords.iterator();
-                while (iterator.hasNext()) {
-                    ConsumerRecord<String, Bytes> record = iterator.next();
-                    handlerMap.get(record.topic()).handle(record);
+            try {
+                ConsumerRecords<String, Bytes> consumerRecords = consumer.poll(Duration.ofMillis(500L));
+                if (!consumerRecords.isEmpty()) {
+                    Iterator<ConsumerRecord<String, Bytes>> iterator = consumerRecords.iterator();
+                    while (iterator.hasNext()) {
+                        ConsumerRecord<String, Bytes> record = iterator.next();
+                        try {
+                            handlerMap.get(record.topic()).handle(record);
+                        } catch (Throwable e) {
+                            log.error("process record error, topic:{}, key:{}, value:{}", record.topic(), record.key(), record.value());
+                            log.error("process record error", e);
```

Comment:
Why need two error lines?

Reply:
Once a record is failed to handle, a portion records of the batch will be discarded.

Reply:
Just want record detail data? what's best practice to do that?

Reply:
Logging data actually isn't useful.